swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/autoscalinggroups/Rules.kt

```diff
@@ -109,3 +111,122 @@ class ZeroInstanceInDiscoveryDisabledServerGroupRule(
     return instance.status == OUT_OF_SERVICE && disabledInDays > disabledDurationInDays
   }
 }
+
+@Component
+class ZeroInstanceRule : ServerGroupRule() {
+  override fun <T : Resource> applyRule(resource: T, ruleDefinition: RuleDefinition?): Result {
+    return apply(resource, ruleDefinition)
+  }
+
+  override fun <T : Resource> apply(resource: T, ruleDefinition: RuleDefinition?): Result {
+    if (resource !is AmazonAutoScalingGroup || !resource.instances.isNullOrEmpty()) {
+      return Result(null)
+    }
+
+    return Result(Summary(description = "Server Group ${resource.resourceId} has no instances.", ruleName = name()))
+  }
+}
+
+@Component
+class ZeroLoadBalancerRule : ServerGroupRule() {
+  override fun <T : Resource> applyRule(resource: T, ruleDefinition: RuleDefinition?): Result {
+    return apply(resource, ruleDefinition)
+  }
+
+  override fun <T : Resource> apply(resource: T, ruleDefinition: RuleDefinition?): Result {
+    if (resource !is AmazonAutoScalingGroup || !resource.loadBalancerNames.isNullOrEmpty()) {
+      return Result(null)
+    }
+
+    return Result(Summary(description = "Server Group ${resource.resourceId} has no load balancer.", ruleName = name()))
+  }
+}
+
+@Component
+class DisabledLoadBalancerRule(
+  private val clock: Clock
+) : ServerGroupRule() {
+  override fun <T : Resource> applyRule(resource: T, ruleDefinition: RuleDefinition?): Result {
+    return apply(resource, ruleDefinition)
+  }
+
+  override fun <T : Resource> apply(resource: T, ruleDefinition: RuleDefinition?): Result {
+    if (resource !is AmazonAutoScalingGroup || resource.isInLoadBalancer()) {
+      return Result(null)
+    }
+
+    val disabledTime = resource.disabledTime()!!
+    val moreThanDays = ruleDefinition?.parameters?.get("moreThanDays") as? Int
+    if (moreThanDays != null && disabledTime.isBefore(LocalDateTime.now(clock).minusDays(moreThanDays.toLong()))) {
+      return Result(
+        Summary(description = "Server Group ${resource.resourceId} has been out of balancer longer than $moreThanDays days.", ruleName = name())
+      )
+    }
+
+    return Result(
```

Comment:
not clear why is this needed ?

Reply:
`moreThanDays` is optional in this rule. This rule would enforce it if present otherwise it just asserts if the server group is disabled

Reply:
Will add that as a comment.

Reply:
gotcha !